### PR TITLE
Generate the Android 7 round app icon

### LINKED ISSUE #23178: AppIcon not displayed for Android 7
### Description

App icon in the launcher and task switcher is not displaying for Android 7. However, Android 8 and up works
Task_Switcher_Android8
AppIcon_Android8
Task_Switcher_Android7
AppIcon_Android7 png

### Steps to Reproduce

Create a new maui app and deploy to an emulator running Android 7 (7.1.1) with a screen resolution of 1280 x 720. Or build and deploy the repro attached to same emulator.

### Link to public reproduction project repository

https://github.com/Syed-RI/DeployTest

### Version with bug

8.0.60 SR6

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

Android 7.1.1

### Did you find any workaround?

Works on Android 8 and up

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by mattleibow**
### Description of Change

This PR fixes the case where Android 7 is very different:

 - Android 6 only supported "square" icons
 - Android 7 added round icons - https://developer.android.com/about/versions/nougat/android-7.1#circular-icons
 - Android 8 switched to adaptive icons - https://developer.android.com/studio/write/create-app-icons#launcher

This means that specifically for Android 7, the round icons are PNG files but for Android 8 they are adaptive icons.

### Issues Fixed

Fixes #23178

### CODE REVIEW

File: src/SingleProject/Resizetizer/src/SkiaSharpAppIconTools.cs [hunk 1]
```diff
@@ -83,6 +83,13 @@ void Draw(SKBitmap tempBitmap, DpiPath dpi, SKSize unscaledCanvasSize)
 
 			using var canvas = new SKCanvas(tempBitmap);
 
+			if (GetClipPath(dpi, canvasSize, unscaledCanvasSize) is { } clipPath)
+			{
+				canvas.Clear(SKColors.Transparent);
+
+				canvas.ClipPath(clipPath, antialias: true);
+			}
```
Comment (mattleibow) [inline src/SingleProject/Resizetizer/src/SkiaSharpAppIconTools.cs:91]: This is the real code change, if there is a clip path attached to the "dpi" then we clip to that first.

File: src/SingleProject/Resizetizer/src/DpiPath.cs [hunk 2]
```diff
@@ -50,11 +61,18 @@ public static DpiPath[] Image
...
 					new DpiPath("mipmap-hdpi", 1.5m, size: new SKSize(48, 48)),
 					new DpiPath("mipmap-xhdpi", 2.0m, size: new SKSize(48, 48)),
 					new DpiPath("mipmap-xxhdpi", 3.0m, size: new SKSize(48, 48)),
 					new DpiPath("mipmap-xxxhdpi", 4.0m, size: new SKSize(48, 48)),
+					// legacy round for Android 7
+					new DpiPath("mipmap-mdpi", 1.0m, "_round", size: new SKSize(48, 48), clip: ClipShape.Circle),
+					new DpiPath("mipmap-hdpi", 1.5m, "_round", size: new SKSize(48, 48), clip: ClipShape.Circle),
+					new DpiPath("mipmap-xhdpi", 2.0m, "_round", size: new SKSize(48, 48), clip: ClipShape.Circle),
+					new DpiPath("mipmap-xxhdpi", 3.0m, "_round", size: new SKSize(48, 48), clip: ClipShape.Circle),
+					new DpiPath("mipmap-xxxhdpi", 4.0m, "_round", size: new SKSize(48, 48), clip: ClipShape.Circle),
```
Comment (mattleibow) [inline src/SingleProject/Resizetizer/src/DpiPath.cs:75]: Just these have a clip shape.

**Review by rmarinho — APPROVED** [review]

**Pull request merged**